docs/data_ingestion.asciidoc

```diff
@@ -0,0 +1,103 @@
+* <<reduce-payload-size>>
+
+[[tune-output-config]]
+[float]
+=== Tune APM Server output parameters for your Elasticsearch cluster
+
+If your Elasticsearch cluster is sized properly,
+but not ingesting the amount of data you expect,
+you can tweak APM Server options to make better use of the cluster:
+
+* Increase `output.elasticsearch.workers` to a number that suits your setup.
+See {ref}/tune-for-indexing-speed.html[tune for indexing speed] for an overview.
+* Ensure `output.elasticsearch.bulk_max_size` is set to a high value, for example 5120.
+  The default of 50 is very conservative.
+* Ensure that `queue.mem.events` is set to a reasonable value compared to your other settings.
+A good rule of thumb is that `queue.mem.events` should equal `output.elasticsearch.worker` multiplied by `output.elasticsearch.bulk_max_size`.
+
+Get detailled information on available <<configuring-output,output configuration options>> and their default values.
+
+[[increase-queue-size]]
+[float]
+=== Increase internal queue size
+
+APM Server uses an internal queue to allow buffering incoming requests until they can be delievered to Elasticsearch. 
+A larger internal queue allows Elasticsearch to be unavailable for longer periods,
+and it alleviates problems that might result from sudden spikes of data.
+You can increase the queue size by overriding `queue.mem.events`.
+Be aware that increasing `queue.mem.events` can significantly affect APM Server memory usage.
+
+[[increase-concurrent-requests]]
+[float]
+=== Increase concurrent requests
+To avoid overfloating the APM Server has an upper limit to how many requests are accepted concurrently.
+This limit is determined by the `apm-server.concurrent_requests` configuration parameter.
+As this limit is set to a rather conservative default you might want to set it to some higher value.
+Be aware though, that increasing the limit can significantly affect APM Server memory usage.
+
+[[add-apm-server-nodes]]
+[float]
+=== Add APM Server nodes
+
+In case the APM Server cannot process incoming requests quickly enough,
+you will see a server timeout.
+
+One way to avoid this problem is to add more processing power to your APM Server cluster.
+This can be easily done by either migrating your APM Server processes to more powerful machines 
+or adding more APM Server nodes.
+Please refer to the <<high-availability, high availability>> section for general information on how to set up multiple APM Server.
+
+[[reduce-payload-size]]
+[float]
+=== Reduce the payload size
+
+Large payloads coming from agents may result in a server timeout.
+You can reduce the payload size by decreasing the `max_queue_size` in the agents.
+This will result in agents sending smaller payloads to the APM Server,
+but the requests will be more frequent.
+See the documentation for the {apm-py-ref}/configuration.html#config-max-queue-size[Python] and {apm-node-ref}/agent-api.html#max-queue-size[Node.js] agents for more information.
```

Comment:
💯 linking here to agent docs! 
should we mention keep-alive? (ie. supported by deserved, disabled by default in most agents)

Reply:
probably this comment is out of place, thou -> can be added in another pr

Reply:
https://github.com/elastic/apm-server/issues/1089